python/test/unit/mesh/test_mesh.py

```diff
@@ -578,3 +579,16 @@ def test_original_index():
     s = sum(mesh.topology.original_cell_index)
     s = MPI.COMM_WORLD.allreduce(s, MPI.SUM)
     assert(s == nx**3 * 6 * (nx**3 * 6 - 1) // 2)
+
+
+def test_boundary_identification():
+    nx = 8
+    mesh = create_unit_square(MPI.COMM_WORLD, nx, nx, ghost_mode=GhostMode.shared_facet)
+    left_cells = locate_entities(mesh, mesh.topology.dim, lambda x: x[0] <= 0.5 + 5e-16)
+    sub_mesh = create_submesh(mesh, mesh.topology.dim, left_cells)
+    sub_mesh[0].topology.create_entities(sub_mesh[0].topology.dim - 1)
+    sub_mesh[0].topology.create_connectivity(sub_mesh[0].topology.dim - 1, sub_mesh[0].topology.dim)
+
+    right_facets = compute_boundary_facets(sub_mesh[0].topology)
```

Comment:
I believe, because of the issue I mentioned above, some facets will be incorrectly marked as being on the boundary on some ranks for a `GhostMode.none` mesh. This would be the case even for a mesh created using `create_unit_square`.